plugins/woocommerce/includes/admin/helper/class-wc-helper-plugin.php

```diff
@@ -33,6 +43,120 @@ public static function is_plugin_active(): bool {
 	 * @return bool
 	 */
 	public static function is_plugin_installed(): bool {
-		return file_exists( WP_PLUGIN_DIR . '/' . self::WOO_MARKETPLACE_PLUGIN_MAIN_FILE );
+		return file_exists( WP_PLUGIN_DIR . '/' . self::WOO_UPDATE_MANAGER_PLUGIN_MAIN_FILE );
+	}
+
+	/**
+	 * Generate the URL to install the Woo Connect plugin.
+	 *
+	 * @return string
+	 */
+	public static function generate_install_url(): string {
+		/**
+		 * Filter the base URL used to install the Woo Update Manager plugin.
+		 *
+		 * @since 8.7.0
+		 */
+		$install_url_base = apply_filters( 'woo_com_base_url', 'https://woo.com/' );
+
+		/**
+		 * Filter the id of the Woo Update Manager plugin.
+		 *
+		 * @since 8.7.0
+		 */
+		$woo_update_manager_plugin_id = apply_filters( 'woo_update_manager_plugin_id', 18734003334043 );
+
+		$install_url = $install_url_base . 'auto-install/step/init/' . $woo_update_manager_plugin_id . '/';
+
+		return self::add_auth_parameters( $install_url );
+	}
+
+	/**
+	 * Add the access token and signature to the provided URL.
+	 *
+	 * @param string $url The URL to add the access token and signature to.
+	 * @return string
+	 */
+	private static function add_auth_parameters( string $url ): string {
+		$auth = WC_Helper_Options::get( 'auth' );
+
+		if ( empty( $auth['access_token'] ) || empty( $auth['access_token_secret'] ) ) {
+			return false;
+		}
+
+		$request_uri  = wp_parse_url( $url, PHP_URL_PATH );
+		$query_string = wp_parse_url( $url, PHP_URL_QUERY );
+
+		if ( is_string( $query_string ) ) {
+			$request_uri .= '?' . $query_string;
+		}
+
+		$data = array(
+			'host'        => wp_parse_url( $url, PHP_URL_HOST ),
+			'request_uri' => $request_uri,
+			'method'      => 'GET',
+		);
+
+		$signature = hash_hmac( 'sha256', wp_json_encode( $data ), $auth['access_token_secret'] );
+
+		return add_query_arg(
+			array(
+				'token'     => $auth['access_token'],
+				'signature' => $signature,
+			),
+			$url
+		);
+	}
```

Comment:
This code is almost identical to the first half of the `\WC_Helper_API::_authenticate` method. How about making it possible to re-use that code instead of adding duplication?

Reply:
The method can not be reused as it is, since it's created for a n api request. Extracted the code for creating the signature which is common for both places. https://github.com/woocommerce/woocommerce/pull/44279/commits/7a644705e989d39ceae751cfd4335d228dfafa39